anydex/wallet/stellar/xlm_db.py

```diff
@@ -0,0 +1,240 @@
+    """
+
+    def __init__(self, db_path):
+        """
+        :param db_path: full path (including db file)
+        """
+        self.session = initialize_db(db_path)
+
+    def get_wallet_secret(self, wallet_name):
+        """
+        Checks the database for a wallet with the given wallet_name.
+        :param wallet_name: wallet_name to check
+        :return: wallet secret if exists else None
+        """
+        secret = self.session.query(Secret).filter(Secret.name == wallet_name).first()
+        if secret:
+            return secret.secret
+        return None
+
+    def add_secret(self, wallet_name: str, secret: str, address: str):
+        """
+        Add a secret to the database with the given parameters.
+
+        :param wallet_name: name of the wallet
+        :param secret: secret (like private key) of the wallet
+        :param address: address (public key) of the wallet
+        """
+        self.session.add(Secret(name=wallet_name, secret=secret, address=address))
+        self.session.commit()
+
+    def get_outgoing_amount(self, address):
+        """
+        Get the amount of lumens we are sending but is not yet confirmed.
+
+        :return: amount of lumens we are sending
+        """
+        pending_outgoing = self.session.query(func.sum(Payment.amount)) \
+            .join(Transaction,
+                  Payment.transaction_hash == Transaction.hash).filter(
+            Transaction.is_pending.is_(True)).filter(
+            Payment.from_ == address).first()[0]
+
+        return pending_outgoing if pending_outgoing else 0
+
+    def update_db(self, transactions: List[Transaction]):
+        """
+        Update the transactions and payments table with the specified transactions.
+        The payments are derived from the transaction envelope.
+        """
+        pending_txs = self.session.query(Transaction).filter(Transaction.is_pending.is_(True)).all()
+        confirmed_txs = self.session.query(Transaction).filter(Transaction.is_pending.is_(False)).all()
+        for transaction in transactions:
+            if transaction in pending_txs:
+                self._update_transaction(transaction)
+            elif transaction not in confirmed_txs:
+                self._insert_transaction(transaction)
+        self.session.commit()
+
+    def _update_transaction(self, transaction):
+        """
+        Update a pending transaction and it's corresponding payments
+        :param transaction: transaction to update
+        """
+        self.session.query(Transaction).filter(Transaction.hash == transaction.hash).update({
+            Transaction.is_pending: False,
+            Transaction.succeeded: transaction.succeeded,
+            Transaction.date_time: transaction.date_time
+        })
+
+    def _insert_transaction(self, transaction):
+        """
+        Insert a transactions into the database.
+        Todo what about merge account
```

Comment:
Has this TODO been tackled already or is it still something you're yet to work upon?

Reply:
It hasn't been implemented yet, but I decided to either figure it out later or just not do it since It seemed hard to do correctly and I don't think the `merge account` operation would be done a lot.